src/main/java/org/springframework/data/elasticsearch/core/SearchOperations.java

```diff
@@ -81,7 +81,7 @@ default long count(Query query, IndexCoordinates index) {
 	 * @param index the index to run the query against
 	 * @return a page with aggregations
 	 */
-	<T> AggregatedPage<T> queryForPage(Query query, Class<T> clazz, IndexCoordinates index);
+	<T> AggregatedPage<SearchHit<T>> queryForPage(Query query, Class<T> clazz, IndexCoordinates index);
```

Comment:
Related to the question above: Changing the component type to `SearchHit<T>` could make it hard for users to consume the actual result. We might want to add convenience methods to `AggregatedPage` to consume a `List<T>` ( `List<T> unwrap()`?). Just a thought.